wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/rapidreactcommandbot/RobotContainer.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj.examples.rapidreactcommandbot;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Drive;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Intake;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Shooter;
+import edu.wpi.first.wpilibj.examples.frisbeebot.Constants.OIConstants;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Storage;
+
+import static edu.wpi.first.wpilibj2.command.CommandGroupBase.parallel;
+
+/**
+ * This class is where the bulk of the robot should be declared. Since Command-based is a
+ * "declarative" paradigm, very little robot logic should actually be handled in the {@link Robot}
+ * periodic methods (other than the scheduler calls). Instead, the structure of the robot (including
+ * subsystems, commands, and button mappings) should be declared here.
+ */
+public class RobotContainer {
+  // The robot's subsystems
+  private final Drive m_robotDrive = new Drive();
+  private final Intake m_intake = new Intake();
+  private final Storage m_storage = new Storage();
+  private final Shooter m_shooter = new Shooter();
+
+  // The driver's controller
+  CommandXboxController m_driverController = new CommandXboxController(OIConstants.kDriverControllerPort);
+
+  public void configureDefaultCommands() {
+    // Configure default commands
+    // Set the default drive command to split-stick arcade drive
+    m_robotDrive.setDefaultCommand(
+        m_robotDrive.arcadeDriveCommand(m_driverController::getLeftY, m_driverController::getRightX));
+  }
+
+  public void configureEventBindings() {
```

Comment:
No actual reason to split this method. Perhaps rename the other one to something that doesn't suggest it's limited to buttons.

Reply:
there's no reason to bind buttons during autonomous, but there's very possibly reason to bind general events.  I think it's good to keep these separate.

Reply:
If both are called from `robotInit`, there's no difference.
However, due to lifetime issues I think it's better to bind everything on startup.
On the other hand, explicit EventLoop usage would be required to separate teleop bindings from test mode bindings -- there should be an example showing such usage, I don't know whether it should be this one.

Reply:
They're not; one of them is called in `teleopInit`

Reply:
I think that binding on startup would be better. Otherwise you either get duplicate bindings that interfere with one another, or you clear everything -- including the event bindings.
Also, it would also be a good way to show that "buttons are events".

Reply:
There's no reason to bind button events on startup when you're not going to ever see a button event until teleop.  I don't see any duplicate bindings here at all - what are you talking about in particular?

Reply:
If bindings are done at `teleopInit`, what happens when the robot is enabled, disabled, and then enabled again? This won't happen in a match, but needing to restart robot code before every enable during practice would be irritating.